src/public/video.ts

```diff
@@ -0,0 +1,133 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from './constants';
+import { registerHandler } from '../internal/handlers';
+/**
+ * Namespace to video extensibility of the SDK.
+ *
+ */
+export namespace video {
+  /**
+   * Represents a video frame.
+   */
+  export interface VideoFrame {
+    /**
+     * Video frame width.
+     */
+    width: number;
+    /**
+     * Video frame height.
+     */
+    height: number;
+    /**
+     * Video frame buffer
+     */
+    data: Uint8ClampedArray;
+    /**
+     * NV12 luma stride, valid only when video frame format is NV12
+     */
+    lumaStride?: number;
+    /**
+     * NV12 chroma stride, valid only when video frame format is NV12
+     */
+    chromaStride?: number;
+    /**
+     * RGB stride, valid only when video frame format is RGB
+     */
+    stride?: number;
+  }
+  /**
+   * Video frame format enum, currentlyl only support NV12
+   */
+  export enum VideoFrameFormat {
+    NV12,
+  }
+  /**
+   * Video frame configuration supplied to Teams to customize the generated video frame parameters, like format.
+   */
+  export interface VideoFrameConfig {
+    /**
+     * video format
+     */
+    format: VideoFrameFormat;
+  }
+
+  /**
+   *  Video effect change type enum
+   */
+  export enum EffectChangeType {
+    /**
+     * current video effect changed.
+     */
+    EffectChanged,
+    /**
+     * disable the video effect
+     */
+    EffectDisabled,
+  }
+
+  /**
+   *  Video frame call back funtion definition
+   */
+  type VideoFrameCallback = (
+    frame: VideoFrame,
+    notifyVideoFrameProcessed: () => void,
+    notifyError: (errorMessage: string) => void,
+  ) => void;
+
+  /**
+   *  Video effect change call back funtion definition
+   */
+  type VideoEffectCallBack = (effectId: string | undefined) => void;
+
+  /**
+   * register to read the video frames in Permissions section.
+   */
+  export function registerForVideoFrame(frameCallback: VideoFrameCallback, config: VideoFrameConfig): void {
+    ensureInitialized(FrameContexts.sidePanel);
+    registerHandler('videoApp.newVideoFrame', (videoFrame: VideoFrame) => {
+      if (videoFrame !== undefined) {
+        frameCallback(videoFrame, notifyVideoFrameProcessed, notifyError);
+      }
+    });
+    sendMessageToParent('videoApp.registerForVideoFrame', [config]);
```

Comment:
Please change `videoApp` to `video` in the SDK action names. These should match the namespace and function name in the SDK.

Reply:
Yes for pointing it out. will rename it.